### PR TITLE
Fix sending StartServiceACK when try to start session with encryption ON

### PR DESCRIPTION
**Pull request by istoilovgithub**
Fix enables requested service protection only if
there is an registered application in the current session.

### CODE REVIEW

**Comment by istoilovgithub** [comment]
@okozlovlux Please review

File: src/components/application_manager/src/application_manager_impl.cc [hunk 1]
```diff
@@ -1283,6 +1283,23 @@ ApplicationManagerImpl::GetHandshakeContext(uint32_t key) const {
   }
   return SSLContext::HandshakeContext();
 }
+
+bool ApplicationManagerImpl::CanStartProtectedService(
+    const int32_t& session_key,
+    const protocol_handler::ServiceType& type) const {
```
Comment (okozlovlux) [inline src/components/application_manager/src/application_manager_impl.cc]: @istoilovgithub , think passing by references is redundant

File: src/components/connection_handler/src/connection_handler_impl.cc [hunk 2]
```diff
@@ -1015,6 +1027,21 @@ bool ConnectionHandlerImpl::ProtocolVersionUsed(
   return false;
 }
 
+#ifdef ENABLE_SECURITY
+
+bool ConnectionHandlerImpl::CanStartProtectedService(
+    const int32_t& session_key,
+    const protocol_handler::ServiceType& type) const {
```
Comment (okozlovlux) [inline src/components/connection_handler/src/connection_handler_impl.cc]: @istoilovgithub , think passing by references is redundant

File: src/components/connection_handler/src/connection_handler_impl.cc [hunk 3]
```diff
@@ -285,17 +285,27 @@ uint32_t ConnectionHandlerImpl::OnSessionStartedCallback(
     const uint8_t session_id,
     const protocol_handler::ServiceType& service_type,
     const bool is_protected,
-    uint32_t* hash_id) {
+    uint32_t* hash_id,
+    bool* out_start_protected) {
```
Comment (okozlovlux) [inline src/components/connection_handler/src/connection_handler_impl.cc]: @istoilovgithub , can you pass it by reference?

Comment (LevchenkoS) [inline src/components/connection_handler/src/connection_handler_impl.cc]: @istoilovgithub, do not forget to add `out_` prefix to `hash_id`. And according our coding stile we mostly use references for output parameters, spatially for a `POD`

File: src/components/application_manager/src/application_manager_impl.cc (hunk 1 above)
Comment (istoilovgithub) [inline src/components/application_manager/src/application_manager_impl.cc]: @okozlovlux Passing parameter by const reference prevents for using it as lvalue in the function scope, so I prefer to keep this one as it is.

Comment (okozlovlux) [inline src/components/application_manager/src/application_manager_impl.cc]: @istoilovgithub, both of params or just serviceType?

Comment (istoilovgithub) [inline src/components/application_manager/src/application_manager_impl.cc]: @okozlovlux It applies to all types in general.

File: src/components/connection_handler/src/connection_handler_impl.cc (hunk 3 above)
Comment (istoilovgithub) [inline src/components/connection_handler/src/connection_handler_impl.cc]: @LevchenkoS Fixed with 83a8c22

**Comment by istoilovgithub** [comment]
Rebuild requred

**Pull request merged**